src/web/app/components/comment-box/comment-row/comment-row.component.spec.ts

```diff
@@ -45,4 +60,118 @@ describe('CommentRowComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  describe('ngOnChanges', () => {
+    it('should properly handle visibility settings if originalComment is defined', () => {
+      component.model = {
+        originalComment: {
+          isVisibilityFollowingFeedbackQuestion: true,
+          commentGiver: 'mockCommentGiver',
+          lastEditorEmail: 'mockEditor@example.com',
+          feedbackResponseCommentId: 12345,
+          commentText: 'Mock comment text',
+          showCommentTo: [CommentVisibilityType.GIVER, CommentVisibilityType.INSTRUCTORS],
+          createdAt: new Date().getTime(),
+          lastEditedAt: new Date().getTime(),
+          showGiverNameTo: [],
+        },
+        commentEditFormModel: {
+          commentText: 'Mock comment text for form',
+          isUsingCustomVisibilities: false,
+          showCommentTo: [CommentVisibilityType.GIVER],
+          showGiverNameTo: [CommentVisibilityType.INSTRUCTORS],
+        },
+        isEditing: true,
+      };
+
+      component.ngOnChanges();
+
+      expect(component.visibilityStateMachine).toBeDefined();
+      expect(component.visibilityStateMachine.allowAllApplicableTypesToSee).toBeDefined();
+
+      expect(spyVisibilityStateMachine.allowAllApplicableTypesToSee).toHaveBeenCalledWith();
+
+      expect(spyCommentService.getNewVisibilityStateMachine).toHaveBeenCalledWith([]);
```

Comment:
let's set questionShowResponsesTo in the component, and check that getNewVisibilityStateMachine is called with the appropriate params